src/services.js

```diff
@@ -422,6 +422,17 @@ export class Services {
           'amp-consent')));
   }
 
+  /**
+   * Returns a promise for the geo service or a promise for null if
+   * the service is not available on the current page.
+   * @param {!Window} win
+   * @return {!Promise<?Object<string,string>>}
+   */
+  static geoForOrNull(win) {
+    return (/** @type {!Promise<?Object<string,string>>} */
+      (getElementServiceIfAvailable(win, 'geo', 'amp-geo', true)));
```

Comment:
I believe  `getElementServiceIfAvailable ` returns a promise that will resolve when the geo promise is found. I believe `{!Promise<?Promise<Object<string, string>>}` was returned instead. Please correct me if I understand it incorrectly.

Reply:
I don't think it double promises otherwise getGeo_ in url-replacements-impl.js wouldn't work.

Reply:
That's what I thought but I couldn't find unit test again the getGeo replacement : ) 
Anyway, I am not familiar enough with `getElementServiceIfAvailable`, it is great that it resolves double promises!

Reply:
Just curious. @choumx @rsimha Do you know where we handle double promises?

Reply:
I'll add for that before we turn this live - FWIW it's in examples/amp-geo.amp.html and works.

Reply:
I tracked this down.  If you follow the calls on getElementServiceIfAvailable it returns Promise.resolve(<whatever the service returns>) basically just wrapping to make sure it's promise.  Since in this case I have to waitForBodyPromise().then(() => ... do my stuff I return a promise that gets wrapped in an promise.resolve which is effectively a no-op.  Let me know if this resolves your concern.

Reply:
Got it! Just learnt `Promise.resolve(promise) = promise` Thanks @jpettitt for the explanation!